src/main/java/weizberg/citibike/ui/CitibikeController.java

```diff
@@ -45,8 +49,18 @@ public Waypoint bikeDropOffLocation() {
     }
 
 
-    public void sendRequest() throws IOException {
-        response = serviceFactory.callLambda(request);
+    public Single<CitibikeResponse> sendRequest() throws IOException {
+        return serviceFactory.callLambda(request)
+                .doOnSuccess(this::handleResponse)
+                .doOnError(this::handleError);
```

Comment:
This is not how I taught you to do it.
You're actually calling `handleResponse` twice now. Once here and once on line 31